### ISSUE TITLE
Namespace - Just FluentAssertion

### ISSUE THREAD

**Issue by cheng93** [fluentassertions/fluentassertions.json#28]
https://github.com/fluentassertions/fluentassertions.json/blob/master/Src/FluentAssertions.Json/JsonAssertionExtensions.cs#L5

Should the namespace for that file be just `FluentAssertion`  ?

Then we won't even need this message in the docs

`Be sure to include using FluentAssertions.Json; otherwise false positives may occur.`

**Comment by dennisdoomen** [comment]
I'm not sure. If we do, you might end up have collisions with the main assembly. Either way, changing this now is going to be a breaking change. cc @jnyrup

**Comment by cheng93** [comment]
I believe collisions would only occur if the main library has a conflicting overload.

In this case JToken/Value/Object

**Comment by jnyrup** [comment]
Some observations:

All FluentAssertions extensions currently use a sub namespace, e.g. `FluentAssertions.Json`, `FluentAssertions.Mvc`, etc.

Currently without `using FluentAssertions.Json`
```c#
((JToken)null).Should(); // GenericCollectionAssertions<JToken>
((JValue)null).Should(); // Ambiguous between `ObjectAssertions`, `GenericCollectionAssertions<JToken>` and `ComparableTypeAssertions<JToken>`
((JObject)null).Should(); // `GenericDictionaryAssertions<string, JToken>`
```

I don't think it would harm to unnest the namespace as a breaking change.

**Comment by dennisdoomen** [comment]
> I don't think it would harm to unnest the namespace as a breaking change.

Not sure what you're saying

**Comment by jnyrup** [comment]
@dennisdoomen No, that sentence didn't make an sense.
I don't think moving this extension from `FluentAssertions.Json` to `FluentAssertions` would cause major problems, as none of the three types extended on are from BCL and `JTokenAssertions` extends `ReferenceTypeAssertions`.
On the other hand, all other FA extensions are in sub namespaces, so I assume they "suffer" from the same problem.

**Comment by cheng93** [comment]
I think the other repos could also be moved to `FluentAssertions` namespace.

Note that this is only the extension class containing the `Should()` methods

**Comment by dennisdoomen** [comment]
Yeah, but changing namespace `FluentAssertions.Json` to `FluentAssertions` will most definitely cause breaking changes to existing users of this extension library.
